zmq.hpp

```diff
@@ -2874,6 +2875,57 @@ class timers
 
 #endif // defined(ZMQ_CPP11) && defined(ZMQ_HAVE_TIMERS)
 
+namespace curve {
+
+inline std::pair<std::string, std::string> keypair()
+{
+    char public_key_buffer[41];
+    char secret_key_buffer[41];
+    int rc = zmq_curve_keypair(public_key_buffer, secret_key_buffer);
+    if (rc == -1)
+        throw zmq::error_t();
+    return std::pair{public_key_buffer, secret_key_buffer};
+}
+
+inline std::string public_(const std::string& secret)
+{
+    if (secret.size() != 40)
+        throw std::runtime_error("Invalid secret string size");
+    char public_key_buffer[41];
+    int rc = zmq_curve_public(public_key_buffer, secret.c_str());
+    if (rc == -1)
+        throw zmq::error_t();
+    return public_key_buffer;
+}
+
+namespace z85 {
+
+inline std::string encode(const std::vector<uint8_t>& data)
+{
+    size_t buffer_size = static_cast<double>(data.size()) * 1.25 + 1;
+    char *buffer = new char[buffer_size];
+    auto *result = zmq_z85_encode(buffer, data.data(), data.size());
+    if (result == nullptr)
+        throw zmq::error_t();
+    std::string dest(result);
+    delete [] buffer;
+    return dest;
+}
+
+inline std::vector<uint8_t> decode(const std::string& encoded)
+{
+    size_t dest_size = static_cast<double>(encoded.size()) * 0.8;
```

Comment:
I am a bit afraid of these floating point computations here and in encode.
What about `(encoded.size() * size_t{4}) / size_t{5}`?